modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/AsyncSqlCursorImplTest.java

```diff
@@ -82,7 +83,7 @@ public void testExceptionRollbacksImplicitTx(boolean implicit, QueryTransactionW
         CompletionException t = assertThrows(CompletionException.class, () -> cursor.requestNextAsync(1).join());
 
         CompletableFuture<Void> f = ((NoOpTransaction) txWrapper.unwrap()).rollbackFuture();
-        assertEquals(implicit, f.isDone(), "Implicit transaction should have been rolled back: " + f);
+        assertTrue(f.isDone(), "Implicit transaction should have been rolled back: " + f);
```

Comment:
not only implicit, as i can see ?

Reply:
Implicit transaction is always rollbacked.